src/storage/txn/txn.cpp

```diff
@@ -153,6 +153,14 @@ TxnTableStore *Txn::GetTxnTableStore(TableEntry *table_entry) {
     return iter->second.get();
 }
 
+void Txn::SetTableStore(const String &table_name, SharedPtr<TxnTableStore> txn_table_store) {
+    if (auto iter = txn_tables_store_.find(table_name); iter != txn_tables_store_.end()) {
```

Comment:
Currently, we don't allow cross-table txn. If txn_tables_store isn't empty, exception should be raised.